bindings/go/configuration/extract/v1alpha1/spec/config.go

```diff
@@ -0,0 +1,214 @@
+package spec
+
+import (
+	"fmt"
+	"slices"
+
+	v1 "ocm.software/open-component-model/bindings/go/configuration/generic/v1/spec"
+	"ocm.software/open-component-model/bindings/go/runtime"
+)
+
+const (
+	// ConfigType defines the type identifier for transformation configurations
+	ConfigType = "extract.oci.artifact.ocm.software"
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	scheme.MustRegisterWithAlias(&Config{}, runtime.NewVersionedType(ConfigType, Version))
+}
+
+// Predefined selector keys for layer properties
+const (
+	// LayerIndexKey is the key used to select layers by index
+	LayerIndexKey = "layer.index"
+	// LayerMediaTypeKey is the key used to select layers by media type
+	LayerMediaTypeKey = "layer.mediaType"
+)
+
+// LayerSelectorOperator represents the operator for selection expressions.
+type LayerSelectorOperator string
+
+const (
+	// LayerSelectorOpIn - the value is in the set of values
+	LayerSelectorOpIn LayerSelectorOperator = "In"
+	// LayerSelectorOpNotIn - the value is not in the set of values
+	LayerSelectorOpNotIn LayerSelectorOperator = "NotIn"
+	// LayerSelectorOpExists - the key exists
+	LayerSelectorOpExists LayerSelectorOperator = "Exists"
+	// LayerSelectorOpDoesNotExist - the key does not exist
+	LayerSelectorOpDoesNotExist LayerSelectorOperator = "DoesNotExist"
+)
+
+// LayerSelectorRequirement represents a single requirement for layer selection.
+// +k8s:deepcopy-gen=true
+type LayerSelectorRequirement struct {
+	// Key is the property key that the selector applies to.
+	// Can be a custom annotation key or predefined keys like layer.index, layer.mediaType.
+	Key string `json:"key"`
+	// Operator represents the relationship between the key and values.
+	Operator LayerSelectorOperator `json:"operator"`
+	// Values is an array of string values. If the operator is In or NotIn,
+	// the value array must be non-empty. If the operator is Exists or DoesNotExist,
+	// the value array must be empty.
+	Values []string `json:"values,omitempty"`
+}
+
+// LayerSelector allows selecting layers based on index, mediatype, and annotations.
+// +k8s:deepcopy-gen=true
+type LayerSelector struct {
+	// MatchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels
+	// map is equivalent to an element of matchExpressions, whose key field is "key", the
+	// operator is "In", and the value array contains only "value". *Note* this is called
+	// match_Labels_ to be in-line with other, more common, matching selector operations.
+	// Though technically, no labels are matched.
```

Comment:
Then lets not call it MatchLabels. I think in OCI, we call manifest addins annotations. https://github.com/opencontainers/image-spec/blob/main/descriptor.md#properties

so that would mean the match happens based on that.

Reply:
Cool! Will do.

Reply:
I also added annotations into the mix and added a BUNCH of tests. Like, after the skeleton was done it was just copy&paste. :D